LoRDeckCodes_py3/lor_deck_encoder.py

```diff
@@ -0,0 +1,243 @@
+from base64 import b32encode, b32decode
+
+from card_code_and_count import CardCodeAndCount
+import varint_translator
+
+CARD_CODE_LENGTH = 7
+faction_to_int = {"DE": 0, "FR": 1, "IO": 2, "NX": 3, "PZ": 4, "SI": 5}
+int_to_faction = {0: "DE", 1: "FR", 2: "IO", 3: "NX", 4: "PZ", 5: "SI"}
+MAX_KNOWN_VERSION = 1
+
+
+def get_deck_from_code(code):
+    result = []
+
+    try:
+        bytes_ = b32decode(code + ('=' * (56 - len(code))))
+    except:
+        raise ValueError("Invalid deck code")
+
+    byte_list = bytearray(bytes_)
+
+    # grab format and version
+    format_ = bytes_[0] >> 4
+    version = bytes_[0] & 15
+    byte_list.pop(0)
+
+    if version > MAX_KNOWN_VERSION:
+        raise ValueError("The provided code requires a higher version \
+                         of this library; please update.")
+
+    for i in range(3, 0, -1):
+        byte_list, num_group_ofs = varint_translator.pop_varint(byte_list)
+
+        for j in range(num_group_ofs):
+            byte_list, num_ofs_in_this_group = varint_translator.pop_varint(
+                                                                     byte_list)
+            byte_list, set_ = varint_translator.pop_varint(byte_list)
+            byte_list, faction = varint_translator.pop_varint(byte_list)
+
+            for k in range(num_ofs_in_this_group):
+                byte_list, card = varint_translator.pop_varint(byte_list)
+
+                set_string = str(set_).zfill(2)
+                faction_string = int_to_faction[faction]
+                card_string = str(card).zfill(3)
+
+                new_entry = CardCodeAndCount(set_string + faction_string
+                                             + card_string, i)
+                result.append(new_entry)
+
+    # the remainder of the deck code is comprised
+    # of entries for cards with counts >= 4.
+    # this will only happen in Limited and special game modes.
+    # the encoding is simply [count] [cardcode]
+    while len(byte_list) > 0:
```

Comment:
in python we can actually just do `while byte_list:` thanks to truthiness https://stackoverflow.com/a/53522

Reply:
Tend to forget about that since this looks more explicit, but can change this and the similar other while in `get_grouped_ofs()`